packages/twenty-server/src/core/quick-actions/quick-actions.service.ts

```diff
@@ -0,0 +1,148 @@
+import { Injectable } from '@nestjs/common';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { Record as IRecord } from 'src/workspace/workspace-query-builder/interfaces/record.interface';
+
+import { isWorkEmail } from 'src/utils/is-work-email';
+import { stringifyWithoutKeyQuote } from 'src/workspace/workspace-query-builder/utils/stringify-without-key-quote.util';
+import { WorkspaceQueryRunnerService } from 'src/workspace/workspace-query-runner/workspace-query-runner.service';
+import { IntelligenceService } from 'src/core/quick-actions/intelligence.service';
+
+@Injectable()
+export class QuickActionsService {
+  constructor(
+    private readonly workspaceQueryRunnunerService: WorkspaceQueryRunnerService,
+    private readonly intelligenceService: IntelligenceService,
+  ) {}
+
+  async createCompanyFromPerson(id: string, workspaceId: string) {
+    const person = (
```

Comment:
Try edge cases like removing the record in an other tab then send the quick action on deleted record, see if it fails gracefully